st2common/st2common/models/api/keyvalue.py

```diff
@@ -111,20 +118,27 @@ def from_model(cls, model, mask_secrets=True):
 
         doc = cls._from_model(model, mask_secrets=mask_secrets)
 
-        if 'id' in doc:
-            del doc['id']
-
-        if model.expire_timestamp:
+        if getattr(model, 'expire_timestamp', None) and model.expire_timestamp:
             doc['expire_timestamp'] = isotime.format(model.expire_timestamp, offset=False)
 
         encrypted = False
-        if model.secret:
+        secret = getattr(model, 'secret', False)
+        if secret:
             encrypted = True
 
-        if not mask_secrets and model.secret:
+        if not mask_secrets and secret:
             doc['value'] = symmetric_decrypt(KeyValuePairAPI.crypto_key, model.value)
             encrypted = False
 
+        scope = getattr(model, 'scope', SYSTEM_SCOPE)
+        if scope:
+            doc['scope'] = scope
+
+        key = doc.get('name', None)
```

Comment:
I just did that for safety. If name isn't present, we won't be able to even write to db so it's fine.